src/kademlia/get_item.cpp

```diff
@@ -88,39 +91,27 @@ void get_item::got_data(bdecode_node const& v,
 		// and it's immutable
 
 		m_data.assign(v);
-		bool put_requested = m_data_callback(m_data, true);
 
-		// if we intend to put, we need to keep going
-		// until we find the closest nodes, since those
-		// are the ones we're putting to
-		if (put_requested)
-		{
+		// There can only be one true immutable item with a given id
```

Comment:
You are right. we couldn't assume other peer always doing right thing, maybe it's a attacker.

For the "powerful feature", what's the meaning of "update", don't put or put same value?
```
	bool put_immutable_item_callback(item& it, boost::function<void()> f)
	bool put_mutable_item_callback(item& it, bool authoritative, boost::function<void(item&)> cb)
```
in dht_tracker always return true, so if client call put_item, we will always do a PUT, even the blob is latest.